cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1355,12 +1355,22 @@ macro(build_snappy)
       "-DCMAKE_INSTALL_PREFIX=${SNAPPY_PREFIX}")
   # Snappy unconditionally enables -Werror when building with clang this can lead
   # to build failures by way of new compiler warnings. This adds a flag to disable
-  # Werror to the very end of the invocation to override the snappy internal setting.
+  # -Werror to the very end of the invocation to override the snappy internal setting.
+  set(SNAPPY_ADDITIONAL_CXX_FLAGS "")
   if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(SNAPPY_ADDITIONAL_CXX_FLAGS "${SNAPPY_ADDITIONAL_CXX_FLAGS} -Wno-error")
+  endif()
+  # Snappy unconditionally disables RTTI, which is incompatible with some other
+  # build settings (https://github.com/apache/arrow/issues/43688).
+  if(NOT MSVC)
+    set(SNAPPY_ADDITIONAL_CXX_FLAGS "${SNAPPY_ADDITIONAL_CXX_FLAGS} -frtti")
+  endif()
+
+  if(NOT MSVC)
```

Comment:
Can we remove this `if()`?

Reply:
No, we cannot, see build error in https://github.com/apache/arrow/actions/runs/10401760097/job/28804899409

Reply:
Ah, https://github.com/apache/arrow/pull/43706/files#r1719156798 will fix it. I'll try it.